internal/http/graphql_client.go

```diff
@@ -0,0 +1,79 @@
+package http
+
+import (
+	"strings"
+
+	"github.com/newrelic/newrelic-client-go/internal/logging"
+	"github.com/newrelic/newrelic-client-go/internal/region"
+	"github.com/newrelic/newrelic-client-go/pkg/config"
+)
+
+// GraphQLClient represents a graphQL HTTP client.
+type GraphQLClient struct {
+	client *NewRelicClient
+	config config.Config
+	logger logging.Logger
+}
+
+// NewGraphQLClient returns a new instance of GraphQLClient.
+func NewGraphQLClient(cfg config.Config) *GraphQLClient {
+	if cfg.BaseURL == "" {
+		cfg.BaseURL = region.NerdGraphBaseURLs[region.Parse(cfg.Region)]
```

Comment:
I realize this is an internal package and not publicly exposed (yet?), but should we try not to use proper nouns in the codebase?

Reply:
Do you mean `NewGraphQLClient` vs `newGraphQLClient`?  The former is required to export the client so it can be used in the `entities` package.

Reply:
I meant `NerdGraphBaseURLs` vs `GraphQLBaseURLs` 😎